scripts/lib/upload-static-assets.ts

```diff
@@ -48,14 +53,22 @@ async function uploadDirectory(prefix: string, directoryPath: string) {
 
 export async function uploadStaticAssets(version: string) {
 	try {
+		await exec('yarn', ['refresh-assets'], { env: { ALLOW_REFRESH_ASSETS_CHANGES: '1' } })
```

Comment:
`setAllVersions` _also_ does:
`await exec('yarn', ['refresh-assets', '--force'], { env: { ALLOW_REFRESH_ASSETS_CHANGES: '1' } })`

so it looks like you'll be doing this twice each time

Reply:
this is mostly here because it does need to have run before we run any of the other code here. it's cached by `lazy` though, so won't run when we call it from a context where it's already done

Reply:
mm, but setAllVersions uses `--force` to skip `lazy`, right?

Reply:
`setAllVersions` runs first, so it's fine if it's `--force`. This one in `uploadStaticAssets` runs after that and isn't `--force` so it should use the cache.

Reply:
PS: this all takes 2s to run, so not really an issue even if it did run twice in some cases
![CleanShot 2024-09-06 at 10 11 27](https://github.com/user-attachments/assets/c803078b-36ae-4fec-b41b-68c8f8c130d5)